### PR TITLE
Adds documentation 

### PR DESCRIPTION
**Pull request by jorgensd**
It is uploaded to github pages when executed on master branch.
Check build artifact for current layout (https://github.com/blocknics/cbc.block/actions/runs/4424330171).
Feedback is appreciated.

I think long-term, we should move your report (the PDF) into the webpage as a separate page.

### CODE REVIEW

File: README.md
Comment (jorgensd) [inline README.md]:
```suggestion
The Python API can be found [below](#api), and an
```

File: README.md
Comment (jorgensd) [inline README.md]:
```suggestion
The API of `CBC-block` can be found here:
```

**Comment by jobh** [comment]
Looks really really good! Even includes the logo, nice touch!

We recently started pushing the coverage report to github pages, just to have it somewhere. I suspect that we need to collect and push all the page artifacts in a single place (workflow), but that should be easy just add the "Download coverage report" from deploy.yaml into a subfolder of the publish dir.

File: README.md
Comment (jobh) [inline README.md:102]: I think we should just have a plain https link to docs, otherwise it looks bad on github,
image

**Comment by jorgensd** [comment]
> Looks really really good! Even includes the logo, nice touch!
> 
> We recently started pushing the coverage report to github pages, just to have it somewhere. I suspect that we need to collect and push all the page artifacts in a single place (workflow), but that should be easy just add the "Download coverage report" from deploy.yaml into a subfolder of the publish dir.

The easiest way to do this would be to the coverage and building in the same workflow.
Otherwise we need to depend on third party actions like: https://github.com/marketplace/actions/download-workflow-artifact

**Comment by jobh** [comment]
> > We recently started pushing the coverage report to github pages, just to have it somewhere. I suspect that we need to collect and push all the page artifacts in a single place (workflow), but that should be easy just add the "Download coverage report" from deploy.yaml into a subfolder of the publish dir.
> 
> The easiest way to do this would be to the coverage and building in the same workflow. Otherwise we need to depend on third party actions like: https://github.com/marketplace/actions/download-workflow-artifact

Yep, basically this step https://github.com/blocknics/cbc.block/blob/master/.github/workflows/deploy.yaml#L63 but with different output directory.

[edit] Coverage is generated by the tests, so they need decoupling by artifact to not get all messy

**Comment by jorgensd** [comment]
Seems like we can use https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

**Comment by jorgensd** [comment]
@jobh I've now made an example of how this would look at: http://jsdokken.com/example-python-package/
with coverage report at:
https://jsdokken.com/example-python-package/code-coverage-report/index.html

with the workflows:
Workflow 1:
https://github.com/jorgensd/example-python-package/blob/main/.github/workflows/test_package.yml
Workflow 2 (dep workflow 1):
https://github.com/jorgensd/example-python-package/blob/main/.github/workflows/build_docs.yml
Workflow 3 (dep workflow 2):
https://github.com/jorgensd/example-python-package/blob/main/.github/workflows/deploy_docs.yml

What do you think?

**Comment by jorgensd** [comment]
Now tested on one of my actual repos:
https://jsdokken.com/dolfinx_mpc/code-coverage-report/index.html
based of:
https://github.com/jorgensd/dolfinx_mpc

**Comment by jobh** [comment]
Hi @jorgensd! It's possible that there are complications that I don't see, but couldn't we simply push a "docs" artifact and upload them together? I'll create a draft PR #14 to illustrate. I haven't tested it so there are probably issues with it, but do you think that would be ok? It's basically the same approach as we have used so far.

**Comment by jorgensd** [comment]
> Hi @jorgensd! It's possible that there are complications that I don't see, but couldn't we simply push a "docs" artifact and upload them together? I'll create a draft PR #14 to illustrate. I haven't tested it so there are probably issues with it, but do you think that would be ok? It's basically the same approach as we have used so far.

Wow, that is really neat. 
I added this in https://github.com/jorgensd/example-python-package/actions/runs/4466300909
And it seems to work like a charm.

https://jsdokken.com/example-python-package/htmlcov/
I'll adapt this PR to do this

**Comment by jobh** [comment]
Cool, thanks! :100:

**Comment by jorgensd** [comment]
@jobh I've now rewritten it to use your original deploy function. Let me know what you think:)

File: .github/workflows/deploy.yaml
Comment (jorgensd) [inline .github/workflows/deploy.yaml]:
```suggestion
```

**Comment by jobh** [comment]
I'm happy :-)

Only thing is the actual link to docs from README, which doesn't render on github. https://github.com/blocknics/cbc.block/tree/dokken/docs#api

Plain HTML link perhaps? Link target doesn't exist until this is merged though, so I'm happy to merge-then-tweak if you prefer.

**Comment by jorgensd** [comment]
> I'm happy :-)
> 
> Only thing is the actual link to docs from README, which doesn't render on github. https://github.com/blocknics/cbc.block/tree/dokken/docs#api
> 
> Plain HTML link perhaps? Link target doesn't exist until this is merged though, so I'm happy to merge-then-tweak if you prefer.

Let's just merge it for now.
One thing we can discuss when you are back is to decouple the intro page to the webpage and the readme for the git repo.
I only did that out of simplicity (it is not required).

**Review by jobh — APPROVED** [review]

**Pull request merged**
